External/FEXCore/Source/Interface/IR/Passes/ConstProp.cpp

```diff
@@ -301,18 +301,29 @@ void ConstProp::LoadMemStoreMemImmediatePooling(IREmitter *IREmit, const IRListV
   for (auto [BlockNode, BlockIROp] : CurrentIR.GetBlocks()) {
     for (auto [CodeNode, IROp] : CurrentIR.GetCode(BlockNode)) {
       if (IROp->Op == OP_LOADMEM || IROp->Op == OP_STOREMEM) {
+        size_t AddrIndex = 0;
+        size_t OffsetIndex = 0;
+
+        if (IROp->Op == OP_LOADMEM) {
+          AddrIndex = IR::IROp_LoadMem::Addr_Index;
+          OffsetIndex = IR::IROp_LoadMem::Offset_Index;
+        }
+        else {
+          AddrIndex = IR::IROp_StoreMem::Addr_Index;
+          OffsetIndex = IR::IROp_StoreMem::Offset_Index;
+        }
         uint64_t Addr;
 
-        if (IREmit->IsValueConstant(IROp->Args[0], &Addr) && IROp->Args[1].IsInvalid()) {
+        if (IREmit->IsValueConstant(IROp->Args[AddrIndex], &Addr) && IROp->Args[OffsetIndex].IsInvalid()) {
```

Comment:
This line actually fixes a bug in the case of `IROp->Op == OP_STOREMEM` here.
Args[1] in LoadMem was Offset, while Args[1] in StoreMem was Value, so in the StoreMem case it was never `IsInvalid()`.

This is now resolved and StoreMem operations will now optimize with an immediate offset rather than never optimizing.